Sources/Common/Communication/EventBusHandler.swift

```diff
@@ -38,7 +41,11 @@ public class CioEventBusHandler: EventBusHandler {
         self.eventCache = eventCache
         self.eventStorage = eventStorage
         self.logger = logger
-        Task { await loadEventsFromStorage() }
+        taskBag += Task { await loadEventsFromStorage() }
```

Comment:
while thinking about this, I think we should assure this is completed, before we add any observer. 

in App lifecycle, that would always be the case, that is when we intialize the SDK, it would be called right away and then `addObserver` when we add a module. 

but i think we should still for good practise sake, store reference of this task and check if its done before adding observer. Something like, 

```
    private var initializationTask: Task<Void, Never>? = nil

// in init
initializationTask = Task { await loadEventsFromStorage() }
```

and then `addObserver` becomes like this

```
taskBag += Task {
            // Await the completion of the initialization task before proceeding.
            await self.initializationTask?.value
            await eventBus.addObserver(eventType.key, action: adaptedAction)
            await replayEvents(forType: eventType, action: adaptedAction)
        }
```

what do you think? i can do it in separate PR if that helps